### PR TITLE
[CO-446] Fix Fee Sanity Check

### PR DESCRIPTION
**Pull request by KtorZ**
## Description

When solving CSL-2526, we introduced an extra sanity check to make the API fails if it was to generate a transaction with too many fees (to prevent us from creating illed transactions because of a programming error in the coin selection / fee calculation).
Yet, it turns out this check is a bit too strict / wrong for it compare transaction fees to an absolute value. Therefore, some transactions with many inputs and, actually expensive fees, fail to be created despite being totally valid.

Backport of [cardano-wallet #199](https://github.com/input-output-hk/cardano-wallet/pull/199)

## Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [x] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: wallet-new/test/unit/Test/Spec/CoinSelection.hs [hunk 1]
```diff
@@ -63,15 +64,9 @@ import           Test.Spec.CoinSelection.Generators (InitialBalance (..),
...
-freeLunchCheck :: Core.Coin -> Bool
-freeLunchCheck c = c == Core.mkCoin 0
-
 -- | The smallest fee possible.
 minFee :: Int -> NonEmpty Core.Coin -> Core.Coin
-minFee _ _ = Core.mkCoin 1
-
-minFeeCheck :: Core.Coin -> Bool
-minFeeCheck c = c == Core.mkCoin 1
+minFee _ outs = Core.mkCoin (fromIntegral $ NE.length outs)
```
Comment (akegalj) [inline wallet-new/test/unit/Test/Spec/CoinSelection.hs]: :+1:

**Comment by rvl** [comment]
Problems with slow macOS builders.

bors r+

**Pull request merged**